recycler-example/src/main/java/com/elpassion/android/commons/recycler_example/group/StableMutableRecyclerWithSectionsActivity.kt

```diff
@@ -21,25 +22,37 @@ class StableMutableRecyclerWithSectionsActivity : AppCompatActivity() {
         setContentView(R.layout.recycler_view_with_action)
         recyclerView.layoutManager = LinearLayoutManager(this)
         val users = createManyUsers()
-        val adapters = users.groupBy(User::organization).mapValues { it.value.map(::StableUserItemAdapter) }
+        val adapters = getAdapters(users)
         val itemsStrategy = CachedMapItemsStrategy(MutableMapItemsStrategy(adapters))
         val adapterCompositor = stableSectionedRecyclerViewAdapter(itemsStrategy)
         recyclerView.adapter = adapterCompositor
 
         clearSectionButton.setOnClickListener {
-            itemsStrategy.set("A", emptyList())
-            adapterCompositor.notifyDataSetChanged()
-            restoreSectionButton.isEnabled = true
-            clearSectionButton.isEnabled = false
+            onClearClicked(adapterCompositor, itemsStrategy)
         }
         restoreSectionButton.setOnClickListener {
-            itemsStrategy.addAll("A", createUsersWithASection().map(::StableUserItemAdapter))
-            adapterCompositor.notifyDataSetChanged()
-            restoreSectionButton.isEnabled = false
-            clearSectionButton.isEnabled = true
+            orRestoreClicked(adapterCompositor, itemsStrategy)
```

Comment:
Generally speaking it is ok to extract this methods but should be done in different commit.